src/mint.rs

```diff
@@ -568,91 +528,78 @@ mod tests {
             content: gen_dbc_content,
-            transaction: gen_dbc_trans,
-            transaction_sigs: BTreeMap::from_iter(vec![(
+            transaction: gen_dbc_tx,
+            transaction_sigs: BTreeMap::from_iter([(
                 GENESIS_DBC_INPUT,
                 (genesis_key, genesis_sig),
             )]),
         };
+        let gen_dbc_name = genesis_dbc.name();
 
-        let inputs = HashSet::from_iter(vec![genesis_dbc.clone()]);
-        let input_hashes = BTreeSet::from_iter(vec![genesis_dbc.name()]);
-
-        let genesis_secrets =
+        let genesis_secret_amounts =
             DbcHelper::decrypt_amount_secrets(&genesis_owner, &genesis_dbc.content)?;
-        let outputs_owner = crate::bls_dkg_id();
 
-        let transaction = ReissueTransaction {
-            inputs: inputs.clone(),
-            outputs: HashSet::from_iter(vec![DbcContent::new(
-                input_hashes.clone(),
-                1000,
-                0,
-                outputs_owner.public_key_set.public_key(),
-                genesis_secrets.blinding_factor,
-            )
-            .unwrap()]),
-        };
+        let output_owner = crate::bls_dkg_id();
+        let (reissue_tx, _output_owners) = crate::TransactionBuilder::default()
+            .add_input(genesis_dbc.clone(), genesis_secret_amounts.clone())
+            .add_output(crate::Output {
+                amount: 1000,
+                owner: output_owner.public_key_set.public_key(),
+            })
+            .build()?;
 
         let sig_share = genesis_node
             .key_manager
-            .sign(&transaction.blinded().hash())?;
+            .sign(&reissue_tx.blinded().hash())?;
 
         let sig = genesis_node
             .key_manager
             .public_key_set()?
             .combine_signatures(vec![sig_share.threshold_crypto()])?;
 
         let reissue_req = ReissueRequest {
-            transaction,
-            input_ownership_proofs: HashMap::from_iter(vec![(
-                genesis_dbc.name(),
-                (genesis_node.key_manager.public_key_set()?.public_key(), sig),
-            )]),
+            transaction: reissue_tx,
+            input_ownership_proofs: HashMap::from_iter([(gen_dbc_name, (genesis_key, sig))]),
         };
 
-        let (t, s) = genesis_node.reissue(reissue_req, input_hashes.clone())?;
+        let (t, s) = genesis_node.reissue(reissue_req, BTreeSet::from_iter([gen_dbc_name]))?;
 
-        let double_spend_transaction = ReissueTransaction {
-            inputs,
-            outputs: HashSet::from_iter(vec![DbcContent::new(
-                input_hashes.clone(),
-                1000,
-                0,
-                outputs_owner.public_key_set.public_key(),
-                genesis_secrets.blinding_factor,
-            )
-            .unwrap()]),
-        };
+        let (double_spend_reissue_tx, _output_owners) = crate::TransactionBuilder::default()
+            .add_input(genesis_dbc, genesis_secret_amounts)
```

Comment:
They are secrets (plural) related to the amount (singular).

```suggestion
            .add_input(genesis_dbc, genesis_amount_secrets)
```